Tests/Integration/tests/MessengerTest.php

```diff
@@ -0,0 +1,169 @@
+
+use Presta\SitemapBundle\Messenger\DumpSitemapMessage;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Messenger\EventListener\StopWorkerOnMessageLimitListener;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Transport\InMemoryTransport;
+use Symfony\Component\Messenger\Worker;
+
+class MessengerTest extends SitemapTestCase
+{
+    private const PUBLIC_DIR = __DIR__ . '/../public';
+
+    protected function setUp(): void
+    {
+        if (!class_exists(MessageBusInterface::class)) {
+            $this->markTestSkipped('Skipping messenger tests, because it is not installed.');
+
+            return;
+        }
+
+        foreach (glob(self::PUBLIC_DIR . '/sitemap.*') as $file) {
+            if (!@unlink($file)) {
+                throw new \RuntimeException('Cannot delete file ' . $file);
+            }
+        }
+    }
+
+    private function index(): string
+    {
+        return self::PUBLIC_DIR . '/sitemap.xml';
+    }
+
+    private function section(string $name, bool $gzip = false): string
+    {
+        return self::PUBLIC_DIR . '/' . $this->sectionFile($name, $gzip);
+    }
+
+    private function sectionFile(string $name, bool $gzip = false): string
+    {
+        return 'sitemap.' . $name . '.xml' . ($gzip ? '.gz' : '');
+    }
+
+    private function fileContent(string $file, bool $gzip = false): string
+    {
+        if ($gzip === false) {
+            return file_get_contents($file);
+        }
+
+        $resource = @gzopen($file, 'rb', false);
+        if (!$resource) {
+            throw new \RuntimeException();
+        }
+
+        $data = '';
+        while (!gzeof($resource)) {
+            $data .= gzread($resource, 1024);
+        }
+        gzclose($resource);
+
+        return $data;
+    }
+
+    public function gzip(): array
+    {
+        return [
+            [false],
+            [true],
+        ];
+    }
+
+    /**
+     * @dataProvider gzip
+     */
+    public function testDumpSitemapUsingMessenger(bool $gzip)
+    {
+        self::bootKernel();
+
+        $index = $this->index();
+        self::assertFileNotExists($index, 'Sitemap index file does not exists before dump');
+
+        $static = $this->section('static', $gzip);
+        self::assertFileNotExists($static, 'Sitemap "static" section file does not exists before dump');
+
+        $blog = $this->section('blog', $gzip);
+        self::assertFileNotExists($blog, 'Sitemap "blog" section file does not exists before dump');
+
+        $archives = $this->section('archives', $gzip);
+        $archives0 = $this->section('archives_0', $gzip);
+        self::assertFileNotExists($archives, 'Sitemap "archive" section file does not exists before dump');
+        self::assertFileNotExists($archives0, 'Sitemap "archive_0" section file does not exists before dump');
+
+        /** @var MessageBusInterface $messageBus */
```

Comment:
This is not the kind of thing I was thinking about.

My thought was more like : adding a controller/command in the integration test project that will actually dispatch the message.
That controller/command should be called using same ways as done in Dump & Http tests

Reply:
Hey, I want to double check this with you. For me the way you described is just only that I'd move the dispatching elsewhere but everything would still be the same, so I don't understand why you want this :)

Reply:
The integration suite we introduced aims to test the code "like a developer using the bundle will".

The suite rely on actual/minimal symfony project. We have actual controllers, actual listener...
I'd like your test to be the same : actual controller, dispatching message on actual (synchronous) bus.

The test should call the controller using browserkit, and assert sitemap files are created and filled as expected.
It should be a mix of both the [Http](https://github.com/prestaconcept/PrestaSitemapBundle/blob/master/Tests/Integration/tests/HttpTest.php) for calling your controller & [Cli](https://github.com/prestaconcept/PrestaSitemapBundle/blob/master/Tests/Integration/tests/CliTest.php) for assertion.

Tell me if this is clear enough ?
(writing on mobile is hard, sorry if you have difficulties to read me)

Reply:
Is it the way now how you wanted? :)

Reply:
Is it now the way you wanted? :)

Reply:
Yes this is the way we designed that suite. :)
I believe that the worker part is somehow required, will try to find out an other way later on